drivers/input/input_ft5336.c

```diff
@@ -88,24 +88,32 @@ static int ft5336_process(const struct device *dev)
 	}
 
 	points = (points >> TOUCH_POINTS_POS) & TOUCH_POINTS_MSK;
-	if (points != 0U && points != 1U) {
-		return 0;
-	}
 
-	/* obtain first point X, Y coordinates and event from:
-	 * REG_P1_XH, REG_P1_XL, REG_P1_YH, REG_P1_YL.
-	 */
-	r = i2c_burst_read_dt(&config->bus, REG_P1_XH, coords, sizeof(coords));
-	if (r < 0) {
-		return r;
-	}
+	switch (points) {
+	case 0:
+		/* no touch = no press */
+		pressed = false;
+		break;
+	case 1:
+		/* obtain first point X, Y coordinates and event from:
+		 * REG_P1_XH, REG_P1_XL, REG_P1_YH, REG_P1_YL.
+		 */
+		r = i2c_burst_read_dt(&config->bus, REG_P1_XH, coords, sizeof(coords));
+		if (r < 0) {
+			return r;
+		}
 
-	event = (coords[0] >> EVENT_POS) & EVENT_MSK;
-	row = ((coords[0] & POSITION_H_MSK) << 8U) | coords[1];
-	col = ((coords[2] & POSITION_H_MSK) << 8U) | coords[3];
-	pressed = (event == EVENT_PRESS_DOWN) || (event == EVENT_CONTACT);
+		row = ((coords[0] & POSITION_H_MSK) << 8U) | coords[1];
+		col = ((coords[2] & POSITION_H_MSK) << 8U) | coords[3];
+		event = (coords[0] >> EVENT_POS) & EVENT_MSK;
+		/* Can pressed ever not be true if touch is not zero? */
```

Comment:
probably not but the current code reads for points == 0 exactly for that reason, this is what I get on my unit (it's a `ER-TFT028A2-4`):

```
[00:00:47.878,448] <inf> ft5336: event: 0, row: 111, col: 201                                                            
[00:00:47.903,625] <inf> ft5336: event: 2, row: 115, col: 173                                                            
[00:00:47.928,863] <inf> ft5336: event: 2, row: 120, col: 145                                                            
[00:00:47.954,101] <inf> ft5336: event: 2, row: 126, col: 117                                                            
[00:00:47.979,309] <inf> ft5336: event: 2, row: 127, col: 116                                                            
[00:00:48.004,364] <inf> ft5336: event: 2, row: 127, col: 116                                                            
[00:00:48.029,418] <inf> ft5336: event: 1, row: 127, col: 116                                                            
```

the last read is with points == 0, the coordinate does not get updated and the event is `EVENT_LIFT_UP`.

Is that not how your unit is behaving? I'm somewhat confused by that.

Reply:
On the Adafruit board it claims to be an FT6206 as well, but if touch is zero, the four coordinate bytes read back zero.

Mine is spamming event: 0, row: 0, col: 0 since event 0 is press down.

Reply:
You sill need coords[0] for the top 4 bits of the 12-bit X position. I can make the other change though.

Reply:
Sorry thought that event was in the last byte, yes please, just the variable then, not that it makes much of a difference but no point keeping it just for the debug print.

Reply:
I've made some other changes:
1. Instead of ignoring double touches, always return the first touch point.
2. Test the TOUCH_ID field for invalid touches and discard if set.

Reply:
Nah it's cool, good job, few nitpicks then we can ask few other folks who worked on this to test it on their variants just in case.